core/core.go

```diff
@@ -617,27 +626,32 @@ func (c *Core) readDev(device io.Reader) ([]byte, error) {
 
 func (c *Core) readWriteDev(
 	body []byte,
-	device io.ReadWriter,
+	acquired *session,
 	mode CallMode,
 ) ([]byte, error) {
 
+	acquired.writeMutex.Lock()
 	if mode == CallModeRead {
 		if len(body) != 0 {
 			return nil, errors.New("non-empty body on read mode")
```

Comment:
...and here's a bug caused by that. `read(body="blabla")` will lock the write mutex forever

Reply:
Nice!

Reply:
Done.